bots/pr/src/main/java/org/openjdk/skara/bots/pr/CSRCommand.java

```diff
@@ -96,37 +102,42 @@ public void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInst
-                var relationship = link.relationship();
-                if (relationship.isEmpty() || !relationship.get().equals("csr for")) {
-                    continue;
-                }
-                // Now the issue has a csr link.
-                var csrIssue = link.issue().orElse(null);
-                if (csrIssue == null) {
-                    // The csr link exists but the csr issue doesn't exist.
-                    // We should remind the user to remove the link firstly.
-                    reply.println("the issue for this pull request, [" + jbsIssue.get().id() + "](" + jbsIssue.get().webUrl()
-                            + "), has a invalid CSR link.");
-                    reply.println("So you can't directly indicate that a CSR request is not needed for this pull request. ");
-                    reply.println("Please firstly remove the CSR link and then use the command `/csr unneeded` again.");
-                    return;
-                }
-                var resolution = csrIssue.properties().get("resolution");
-                if (resolution == null || resolution.isNull()
-                        || resolution.get("name") == null || resolution.get("name").isNull()
-                        || csrIssue.state() != Issue.State.CLOSED
-                        || !resolution.get("name").asString().equals("Withdrawn")) {
-                    // The issue has a non-withdrawn csr issue, the bot should direct the user to withdraw the csr firstly.
-                    reply.println("the issue for this pull request, [" + jbsIssue.get().id() + "](" + jbsIssue.get().webUrl() + "), has " +
-                            "a non-withdrawn CSR request: [" + csrIssue.id() + "](" + csrIssue.webUrl() + "). ");
-                    reply.println("So you can't directly indicate that a CSR request is not needed for this pull request. ");
-                    reply.println("Please firstly withdraw the CSR request: [" + csrIssue.id() + "](" + csrIssue.webUrl() + "), "
-                            + "and then use the command `/csr unneeded` again.");
-                    return;
-                }
+
+            var csrLink = csrLink(jbsIssue.get());
+            if (csrLink.isEmpty()) {
+                // The issue has no csr link, the bot should just remove the csr label.
+                pr.removeLabel(CSR_LABEL);
+                reply.println("determined that a [CSR](https://wiki.openjdk.java.net/display/csr/Main) request " +
+                        "is not needed for this pull request.");
+                return;
             }
-            // The issue has no csr or the csr has been withdrawn, the bot should just remove the csr label.
+
+            var csrOptional = csrLink.flatMap(Link::issue);
+            if (csrOptional.isEmpty()) {
+                // The csr link exists but the csr issue doesn't exist.
+                // We should remind the user to remove the link firstly.
```

Comment:
I missed this in the previous patch.

I don't think we can ask the user to remove a broken link in JBS. I'm pretty sure that would require admin actions. Here we should just treat it as if the link didn't exist and remove the label.

Reply:
Fixed.